### PR TITLE
Basic component documentation continued

### PR DESCRIPTION
**Pull request by badnames**
Finish remaining tasks in #333.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Review by absurd-turtle — APPROVED** [review]
LGTM

**Review by temmey — APPROVED** [review]

File: frontend/src/components/Form/PaginatedSelectMenu.tsx [hunk 1]
```diff
@@ -41,13 +41,19 @@ export interface PaginatedSelectMenuProps<
   required?: boolean;
   /** Text that is displayed instead of the input if it has not been selected yet.*/
   placeholder?: string;
+  /** Maximum time interval between two inputs (in milliseconds) before they are considered separate inputs. */
   debounceTimeout?: number;
+  /** Options controlling how data is loaded by react-select. */
```
Comment (markus2330) [inline frontend/src/components/Form/PaginatedSelectMenu.tsx:46]: Can you add a link to more information about this?

File: frontend/src/components/Form/PaginatedSelectMenu.tsx [hunk 2]
```diff
@@ -41,13 +41,19 @@ export interface PaginatedSelectMenuProps<
...
   loadOptions: LoadOptions<Option, GroupBase<Option>, PageAdditionalInfo>;
-  handleOptionsChange?: (option: unknown) => void;
   /** Callback that is invoked every time the user selects a new option. The single argument represents the selected option.*/
+  handleOptionsChange?: (option: unknown) => void;
+  /** This callback gets executed if any input was made by the user */
   onChange?: () => void;
 }
 
+/**
+ * Like SelectMenu, but with added pagination using react-select-async-paginate.
```
Comment (markus2330) [inline frontend/src/components/Form/PaginatedSelectMenu.tsx:55]: before "but" is only a comma if both parts are full sentences.

```suggestion
 * Like SelectMenu but with added pagination using react-select-async-paginate.
```

File: frontend/src/components/Form/SelectMenu.tsx
Comment (markus2330) [inline frontend/src/components/Form/SelectMenu.tsx:1]: I only see CreatableSelectMenu but not SelectMenu in my Storybook?

File: frontend/src/components/Form/SelectMenu.tsx [hunk 3]
```diff
@@ -10,22 +10,38 @@ export interface SelectMenuProps<
...
   ) => void;
+  /** Callback that is invoked if the user made any input. */
   onChange?: () => void;
+  /** Callback that is invoked every time the user changed the search query. */
   onInputChange?: (inputValue: string) => void;
 }
 
+/**
+ * Allows the user to choose from a set of predetermined options.
+ * Based on react-select.
```
Comment (markus2330) [inline frontend/src/components/Form/SelectMenu.tsx:43]: Please also add a link here.

File: frontend/src/components/Header/PageTitle.tsx
Comment (markus2330) [inline frontend/src/components/Header/PageTitle.tsx:1]: I also don't see PageTitle in Storybook?

File: doc/ui_components/Modal/ModalContainer.md
Comment (markus2330) [inline doc/ui_components/Modal/ModalContainer.md:1]: I cannot find ModalContainer in my Storybook?

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job! Unfortunately I cannot find a few components in my storybook. Am I doing something wrong or is there something needed so that they are visible there?

File: frontend/src/components/Form/SelectMenu.tsx
Comment (badnames) [reply frontend/src/components/Form/SelectMenu.tsx:1]: I forgot that the storybook files also need to be added.
I will fix it ASAP.

File: frontend/src/components/Form/SelectMenu.tsx (hunk 3 above)
Comment (badnames) [reply frontend/src/components/Form/SelectMenu.tsx:43]: See above.

File: frontend/src/components/Form/PaginatedSelectMenu.tsx (hunk 1 above)
Comment (badnames) [reply frontend/src/components/Form/PaginatedSelectMenu.tsx]: I could not find a fitting link that would explain this type.
However I have clarified the documentation a bit.

File: frontend/src/components/Form/SelectMenu.tsx [hunk 4]
```diff
@@ -10,22 +10,38 @@ export interface SelectMenuProps<
...
   ) => void;
+  /** Callback that is invoked if the user made any input. */
   onChange?: () => void;
+  /** Callback that is invoked every time the user changed the search query. */
   onInputChange?: (inputValue: string) => void;
 }
 
+/**
+ * Allows the user to choose from a set of predetermined options.
+ * Based on react-select (https://react-select.com/home).
```
Comment (markus2330) [inline frontend/src/components/Form/SelectMenu.tsx:43]: Please use proper link here that can be clicked.

File: frontend/src/components/Header/PageTitle.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  component: PageTitle,
+};
+
+export default meta;
+
+type Story = StoryObj<typeof PageTitle>;
+
+export const PageTitleStory: Story = {
+  args: {
+    title: 'Praise Permaculture!',
```
Comment (markus2330) [inline frontend/src/components/Header/PageTitle.stories.tsx:15]:
```suggestion
    title: 'I love Permaculture!',
```

**Review by markus2330 — APPROVED** [review]
Thx, great job! Good to see this cleanup+more docu! :heart:

**Pull request merged**